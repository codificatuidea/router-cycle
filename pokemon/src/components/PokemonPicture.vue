<template>
  <div class="pokemon-container">
    <img class="hidden-pokemon" :src="imgSrc" />
    <img v-if="showPokemon" class="fade-in" :src="imgSrc" />
  </div>
</template>

<script>
export default {
  props: {
    pokemonId: {
      type: Number,
      required: true
    },
    showPokemon: {
      type: Boolean,
      required: true,
      default: false
    }
  },
  computed: {
    imgSrc() {
      return `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/${this.pokemonId}.png`
    }
  }
}
</script>

<style>
.pokemon-container {
    height: 200px;
}
img {
  height: 200px;
  position: absolute;
  margin-left: -70px;
  user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-user-drag: none;
  -webkit-user-select: none;
}
.hidden-pokemon {
    filter: brightness(0);
}
</style>