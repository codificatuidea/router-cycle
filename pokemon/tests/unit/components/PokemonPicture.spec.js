import { shallowMount } from '@vue/test-utils'
import PokemonPicture from '@/components/PokemonPicture'

describe('PokemonPicture component', () => {

    test('Should do match with snapshot', () => {
        const wrapper = shallowMount(PokemonPicture, {
            props: {
                pokemonId: 1,
                showPokemon: false
            }
        })
        expect(wrapper.html()).toMatchSnapshot()
    })

    test.only('Should exist hidden image and pokemon number 100', () => {
        const pokemonId = 100
        const wrapper = shallowMount(PokemonPicture, {
            props: {
                pokemonId,
                showPokemon: false
            }
        })
        const [imgHidden, imgFadeIn] = wrapper.findAll('img')
        expect(imgHidden.classes('hidden-pokemon')).toBeTruthy()
        expect(imgHidden.exists()).toBeTruthy()
        expect(imgHidden.attributes('src')).toBe(`https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/${pokemonId}.png`)
        expect(imgFadeIn).toBeUndefined()

    })
    // props: { pokemonId: 100, showPokemon: false }
    // https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/100.png

    test.only('Should exist fade-in image and pokemon number 100 if showPokemon is true', () => {
        const pokemonId = 100
        const wrapper = shallowMount(PokemonPicture, {
            props: {
                pokemonId,
                showPokemon: true
            }
        })
        const [_, imgFadeIn] = wrapper.findAll('img')
        expect(imgFadeIn.exists()).toBeTruthy()
        expect(imgFadeIn.classes('fade-in')).toBeTruthy()
        expect(imgFadeIn.attributes('src')).toBe(`https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/${pokemonId}.png`)
    })
    // props: { pokemonId: 100, showPokemon: true }
    // https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/100.png

    
})