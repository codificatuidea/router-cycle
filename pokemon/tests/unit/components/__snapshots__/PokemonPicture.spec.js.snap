// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPicture component Should do match with snapshot 1`] = `
<div class="pokemon-container"><img class="hidden-pokemon" src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/1.png">
  <!--v-if-->
</div>
`;
